src/parser/deserializer.ts

```diff
@@ -599,12 +599,22 @@ function deserializeObject(
       index = index + 12;
 
       // Upgrade to DBRef type
-      object[name] = new DBRef(namespace, oid);
+      value = new DBRef(namespace, oid);
     } else {
       throw new Error(
         'Detected unknown BSON type ' + elementType.toString(16) + ' for fieldname "' + name + '"'
       );
     }
+    if (name === '__proto__') {
+      Object.defineProperty(object, name, {
+        value,
+        writable: true,
+        enumerable: true,
+        configurable: true
+      });
+    } else {
```

Comment:
What's the reason for having an else for the non-proto cases? Are there special properties that the above block would not handle in the way we want?

Reply:
@dariakp This was a performance concern that @nbbeeken and I talked about in the ticket – I did benchmark this, `Object.defineProperty()` is (somewhat understandably) significantly slower than just setting the property directly.

Reply:
Oh that's interesting, I saw the note on the ticket saying that we weren't sure whether it was optimized by V8, but not the actual outcome of any benchmarking. Just out of curiosity, what *was* the performance degradation?

Reply:
It’s about 50 % slower: https://jsbench.me/w7ko1ntaf5/1

I guess it’s fair that V8 doesn’t optimize this as much, it’s not nearly as common for properties to be defined this way.

Reply:
Fair and yeah, that certainly qualifies as significant, thanks for providing the context!